extensions/ql-vscode/src/remote-queries/variant-analysis-view.ts

```diff
@@ -149,6 +150,9 @@ export class VariantAnalysisView
       case "openLogs":
         await this.openLogs();
         break;
+      case "telemetry":
```

Comment:
We'd have to add this to every view. Can we add it in some generic way in `AbstractWebview` instead? (e.g. providing an implementation at the base that is called from all views)? This can be a separate change.

Reply:
I'll have a look into it. I agree we could add this to `AbstractWebview` and therefore have it be implemented for all views for free. This will also protect us against adding telemetry to a shared component and accidentally having it send a message that the view doesn't understand.

Reply:
I've been experimenting with a few different ways of doing this and I can't get the types to work out. I'd like it so that `FromVariantAnalysisMessage` doesn't have to include the `TelemetryMessage` type, and also not have the webview subclass have to call a super method (or at least have it be enforced). I can't get it all to work.

The types just don't like it, because when doing `SomeTemplatedType | TelemetryMessage` it can't tell that the `t` field is unique, and so doing the switch to guarantee enumerating all types doesn't work.

Also, it seems that if a superclass has an abstract method, then the subclass can implement it with different types for the parameters and it doesn't seem to complain 🤷 , which isn't helpful.

My plan currently is to leave it for this PR and look at it again in a followup, where we can get input from people on the various options without delaying this PR.

Reply:
I've been experimenting with a few different ways of doing this and I can't get the types to work out. I'd like it so that `FromVariantAnalysisMessage` doesn't have to include the `TelemetryMessage` type, and also not have the webview subclass have to call a super method (or at least have it be enforced). I can't get it all to work.

The types just don't like it, because when doing `SomeTemplatedType | TelemetryMessage` it can't tell that the `t` field is unique, and so doing the switch to guarantee enumerating all types doesn't work.

Also, it seems that if a superclass has an abstract method, then the subclass can implement it with different types for the parameters and it doesn't seem to complain 🤷 , which isn't helpful.

My plan currently is to leave it for this PR and look at it again in a followup, where we can get input from people on the various options without delaying this PR.

Reply:
> This will also protect us against adding telemetry to a shared component and accidentally having it send a message that the view doesn't understand.

I think we could already have this problem. The views in `common` are also used by the remote queries view, so the remote queries view will throw an error when the `telemetry` message is sent. So, I think we need to need to at least add handling of this message to the remote queries view.

Reply:
I was semi-intentionally ignoring the remote queries view since people shouldn't be using that anymore, but you're right maybe we do need to support that here until we actually remove it for good.

Reply:
I've added the telemetry message to the remote queries view too.